src/components/FeatureViews/Courses/ModalTextEditor.js

```diff
@@ -134,122 +185,309 @@ const ModalTextEditor = ({
         }
       }
     }
-  `;
+  `,
+};
 
-  const [sendEmail, sendEmailResult] = useMutation(SEND_EMAIL, {
-    onCompleted: () => handleClose(false),
-    error: (err) => console.error(err),
-  });
+const QUERY_KEY = {
+  ANNOUNCEMENTS: "announcements",
+  COURSE_SESSIONS: "sessionNotes",
+};
 
-  const [createSessionNote, createSessionNoteResult] = useMutation(
-    CREATE_SESSION_NOTE,
-    {
-      onCompleted: () => handleClose(false),
-      error: (err) => console.error(err),
-      update: (cache, { data }) => {
-        const [newSessionNote] = Object.values(data.createSessionNote);
-        const cachedSessionNote = cache.readQuery({
-          query: GET_SESSION_NOTES,
-          variables: { sessionId: sessionId },
-        })["sessionNotes"];
-        let updatedSessionNotes = [...cachedSessionNote];
-        const matchingIndex = updatedSessionNotes.findIndex(
-          ({ id }) => id === newSessionNote.id
-        );
-        if (matchingIndex === -1) {
-          updatedSessionNotes = [...cachedSessionNote, newSessionNote];
-        } else {
-          updatedSessionNotes[matchingIndex] = newSessionNote;
-        }
-        cache.writeQuery({
-          data: {
-            ["sessionNotes"]: updatedSessionNotes,
-          },
-          query: GET_SESSION_NOTES,
-          variables: { sessionId: sessionId },
-        });
-      },
-    }
-  );
+const MUTATION_KEY = {
+  ANNOUNCEMENTS: "createAnnouncement",
+  COURSE_SESSIONS: "createSessionNote",
+};
 
-  const handleClose = () => handleCloseForm(false);
+const ModalTextEditor = ({
+  open,
+  handleCloseForm,
+  userId,
+  origin,
+  posterId,
+  sessionId,
+  noteId,
+  textSubject,
+  textBody,
+  buttonState,
+  announcementId,
+}) => {
+  const classes = useStyles();
+  const [sendEmailCheckbox, setSendEmailCheckbox] = useState(false);
+  const [sendSMSCheckbox, setSendSMSCheckbox] = useState(false);
+  const [subject, setSubject] = useState(textSubject);
+  const [body, setBody] = useState(() => EditorState.createEmpty());
```

Comment:
Optional: since you call this function on its own and return its value directly, the arrow function is not necessary; you can simply put `useState(EditorState.createEmpty)`, and the function is called as expected.